src/Catalyst.Core.Modules.Web3/Web3EthApiExtensions.cs

```diff
@@ -0,0 +1,51 @@
+using System;
+using Catalyst.Abstractions.Kvm.Models;
+using Catalyst.Abstractions.Ledger;
+using Catalyst.Protocol.Deltas;
+using LibP2P;
+
+namespace Catalyst.Core.Modules.Web3 
+{
+    public static class Web3EthApiExtensions
+    {
+        public static Delta GetLatestDelta(this IWeb3EthApi api)
+        {
+            return GetDelta(api, api.DeltaResolver.LatestDelta);
+        }
+
+        public static Delta GetDelta(this IWeb3EthApi api, Cid cid)
+        {
+            if (!api.DeltaCache.TryGetOrAddConfirmedDelta(cid, out Delta delta))
+            {
+                throw new Exception($"Delta not found '{cid}'");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['System.Exception' should not be thrown by user code.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4585091)